gojmx/types.go

```diff
@@ -56,6 +62,38 @@ func (j *AttributeResponse) GetValue() interface{} {
 	}
 }
 
+// GetValueAsFloat casts the value from AttributeResponse to float based on type.
+func (j *AttributeResponse) GetValueAsFloat() (float64, error) {
+	switch (*j).ResponseType {
+	case ResponseTypeBool:
+		if j.BoolValue {
+			return 1, nil
+		}
+		return 0, nil
+	case ResponseTypeString:
+		parsedValue, err := strconv.ParseFloat(fmt.Sprintf("%v", j.StringValue), 64)
+		if err != nil {
+			return 0, err
+		}
+		if isNaNOrInf(parsedValue) {
+			return 0, ErrNonNumeric
+		}
+		return parsedValue, nil
+	case ResponseTypeDouble:
+		return j.DoubleValue, nil
+	case ResponseTypeInt:
+		return float64(j.IntValue), nil
+	case ResponseTypeErr:
+		return 0, nil
+	default:
+		panic(fmt.Sprintf("unkown value type: %v", j.ResponseType))
```

Comment:
I left this as a panic like it's done in the GetValue, I assume there is a reason for panic here directly

Reply:
This was clarified with CCiutea